api/src/main/java/com/datastrato/gravitino/authorization/Role.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.authorization;
+
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.annotation.Evolving;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The interface of a role. The role is the entity which has kinds of privileges. One role can have
+ * multiple privileges of one entity. Gravitino chooses to grant one entity to one role to avoid
+ * granting too many privileges to one role.
+ */
+@Evolving
+public interface Role extends Auditable {
+
+  /**
+   * The name of the role.
+   *
+   * @return The name of the role.
+   */
+  String name();
+
+  /**
+   * The properties of the group. Note, this method will return null if the properties are not set.
+   *
+   * @return The properties of the role.
+   */
+  Map<String, String> properties();
+
+  /**
+   * The privileges of the role. All privileges belong to one entity. For example: If the entity is
+   * a table, the privileges could be `READ TABLE`, `WRITE TABLE`, etc.
+   *
+   * @return The privileges of the role.
+   */
+  List<Privilege> privileges();
+
+  /**
+   * The privilege entity identifier of the role. For example: If the entity is a table, the
+   * identifier may be `catalog1.schema1.table1`.
```

Comment:
Can it be like "a.b.*", "a.b", “a”? Because if it returns "a", is it a catalog or a table?

Reply:
In my opinion, we don't support a.b.*. If we return "a", it means a catalog.

Reply:
So after the offline discussion, do you still insist on this, or you can support "*"?

Reply:
I decide to return 
`a.b.*` if the entity contains privilege of sub-enties. I will handle it in the logic of operations.

For example,
If `a` has loadCatalog. It will return `a`
If `a` has createSchema. It will return `a.*`.
If `a` has loadSchema. It will return `a.*`.

Reply:
I will correct the comment.